### PR TITLE
test: add upgrade tests

### PR DESCRIPTION
**Pull request by ErinHales**

### CODE REVIEW

File: typescript/optics-tests/test/upgrade.test.ts [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+describe('Upgrade', async () => {
+  let signer: types.Signer,
+    proxy: contracts.MysteryMathV1,
+    upgradeBeacon: contracts.UpgradeBeacon,
+    upgradeBeaconController: contracts.UpgradeBeaconController;
+  const a = 5;
+  const b = 10;
+  const stateVar = 17;
+
+  async function deployProxy(): Promise<any> {
```
Comment (ErinHales) [inline typescript/optics-tests/test/upgrade.test.ts:20]: TODO: update `Promise<any>` with correct type

File: typescript/optics-tests/test/upgrade.test.ts [hunk 2]
```diff
@@ -0,0 +1,112 @@
+import { expect } from 'chai';
+import { ethers } from 'hardhat';
+import * as types from 'ethers';
+import { getTestDeploy } from './testChain';
+import { Updater } from '../lib';
+import * as ProxyUtils from '../../optics-deploy/src/proxyUtils';
+import * as deploys from '../../optics-deploy/src/deployOptics';
+import { Deploy } from '../../optics-deploy/src/chain';
```
Comment (YoduYodu) [inline typescript/optics-tests/test/upgrade.test.ts:8]: line 5,6,8 not used

Comment (YoduYodu) [reply typescript/optics-tests/test/upgrade.test.ts:8]: others look good

Comment (ErinHales) [reply typescript/optics-tests/test/upgrade.test.ts]: Good eye, thanks!

File: typescript/optics-tests/test/upgrade.test.ts (hunk 1 above)
Comment (prestwich) [reply typescript/optics-tests/test/upgrade.test.ts:17]: you could use the `Proxy<>` type from deploys

**Pull request merged**